src/components/MxHeader/getClickHandlerForLinkWithMicroflow.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import { navigateByCallingMicroflow } from '../../utils/mxHelpers';
+
+const getClickHandlerForLinkWithMicroflow = ({
+    microflow,
+    url,
+    predicate,
+    callback,
+}: {
+    microflow: string;
+    url: string;
+    predicate?: () => boolean;
+    callback?: () => void;
+}) => (event: React.MouseEvent<HTMLAnchorElement>) => {
+    if ((predicate && predicate()) || !predicate) {
+        event.preventDefault();
+        navigateByCallingMicroflow(microflow, url);
+        callback && callback();
+    }
+};
+
```

Comment:
Why this utility function is located in the components?

Reply:
It is only used by components in `MxHeader` but not everywhere. I'd put code as close as possible to its usage.

Reply:
Also the file extension is tsx but it has not any react-dom inside.